spec/Spec/KZG.cry

```diff
@@ -108,6 +108,126 @@ verify_kzg_proof commitment_bytes z_bytes y_bytes proof_bytes =
                           (bytes_to_bls_field y_bytes)
                           (bytes_to_kzg_proof proof_bytes)
 
+/**
+ *  Add zeros.
+ */
+padding_zeros : [16][8] -> [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB][8]
+padding_zeros value = result where
+    result = (zero : [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB - 16][8]) # value
+
+/**
+ * Return the Fiat-Shamir challenge required by the rest of the protocol.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_challenge
+ * Discuss with Galois:
+ *      -   Overflow in data = blob + degree_poly + FIAT_SHAMIR_PROTOCOL_DOMAIN_with_leading_zeros + commitment_in_bytes_with_leading_zeros
+ */
+compute_challenge: Blob -> KZGCommitment -> BlsFieldElement
```

Comment:
_Addressing the question about overflow:_ Python uses the `+` operator here to actually mean concatenation of bytes strings, so there is no overflow concern and for Cryptol we need to use `#`, the concatenation operator.

So here is my recommended implementation:
```
compute_challenge: Blob -> KZGCommitment -> BlsFieldElement
compute_challenge blob commitment = hash_to_bls_field data where
    degree_poly = groupBy`{8, 16} `(FIELD_ELEMENTS_PER_BLOB)
    commitment_bytes = groupBy`{8} commitment
    data = FIAT_SHAMIR_PROTOCOL_DOMAIN # degree_poly # blob # commitment_bytes
```